File: ContestOfSharp/TestOfSharp/Challenges/Challenge12Tests.cs
using static ContestOfSharp.Challenges.Challenge12;
namespace TestOfSharp.Challenges;

[TestClass]
public class Challenge12Tests
{
    [TestMethod]
    public void TestOuterRim_Exclusive()
    {
        var result = OuterRim("phone", "mug");
        Assert.AreEqual("phone", result.Item1);
        Assert.AreEqual("mug", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_SomeShared()
    {
        var result = OuterRim("plant", "fan");
        Assert.AreEqual("plt", result.Item1);
        Assert.AreEqual("f", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_SomeSharedAndSomeNot()
    {
        var result = OuterRim("apple", "banana");
        Assert.AreEqual("ple", result.Item1);
        Assert.AreEqual("bn", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_Anagram()
    {
        var result = OuterRim("disney", "sidney");
        Assert.AreEqual("", result.Item1);
        Assert.AreEqual("", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_Empty()
    {
        var result = OuterRim("", "");
        Assert.AreEqual("", result.Item1);
        Assert.AreEqual("", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_OneEmpty()
    {
        var result = OuterRim("hello", "");
        Assert.AreEqual("helo", result.Item1);
        Assert.AreEqual("", result.Item2);
    }
    
    [TestMethod]
    public void TestOuterRim_OneEmptyAndOneShared()
    {
        var result = OuterRim("hello", "world");
        Assert.AreEqual("he", result.Item1);
        Assert.AreEqual("wrd", result.Item2);
    }
}

File: ContestOfSwift/TestsOfSwift/Challenges/Challenge12Tests.swift
//
//  Challenge12Tests.swift
//  TestsOfSwift
//
//  Created by Sergio Daniel on 4/02/25.
//

import XCTest

final class Challenge12Tests: XCTestCase {

    func testOuterRim_exclusive() {
        let result = Challenge12.outerRim(str1: "phone", str2: "mug")
        XCTAssertEqual(result.out1, "phone")
        XCTAssertEqual(result.out2, "mug")
    }
    
    func testOuterRim_someShared() {
        let result = Challenge12.outerRim(str1: "plant", str2: "fan")
        XCTAssertEqual(result.out1, "plt")
        XCTAssertEqual(result.out2, "f")
    }
    
    func testOuterRim_someSharedAndSomeNot() {
        let result = Challenge12.outerRim(str1: "apple", str2: "banana")
        XCTAssertEqual(result.out1, "pple")
        XCTAssertEqual(result.out2, "bnn")
    }
    
    func testOuterRim_anagram() {
        let result = Challenge12.outerRim(str1: "disney", str2: "sidney")
        XCTAssertEqual(result.out1, "")
        XCTAssertEqual(result.out2, "")
    }
    
    func testOuterRim_empty() {
        let result = Challenge12.outerRim(str1: "", str2: "")
        XCTAssertEqual(result.out1, "")
        XCTAssertEqual(result.out2, "")
    }
    
    func testOuterRim_oneEmpty() {
        let result = Challenge12.outerRim(str1: "hello", str2: "")
        XCTAssertEqual(result.out1, "hello")
        XCTAssertEqual(result.out2, "")
    }
    
    func testOuterRim_oneEmptyOneShared() {
        let result = Challenge12.outerRim(str1: "hello", str2: "world")
        XCTAssertEqual(result.out1, "he")
        XCTAssertEqual(result.out2, "wrd")
    }

}
